hamster_lib/helpers/helpers.py

```diff
@@ -59,3 +60,83 @@ def _load_tmp_fact(filepath):
                     content=fact, type=type(fact))
             ))
     return fact
+
+
+def parse_raw_fact(raw_fact):
+    """
+    Extract semantically meaningful sub-components from a ``raw fact`` text.
+
+    Args:
+        raw_fact (text_type): ``raw fact`` text to be parsed.abs
+
+    Returns:
+        dict: dict with sub-components as values.
+    """
+    def at_split(string):
+        """
+        Return everything in front of the (leftest) '@'-symbol, if it was used.
+
+        Args:
+            string (str):  The string to be parsed.
+
+        Returns:
+            tuple: (front, back) representing the substrings before and after the
+                most left ``@`` symbol. If no such symbol was present at all,
+                ``back=None``. Both substrings have been trimmed of any leading
+                and tailing whitespace.
+
+        Note:
+            If our string contains multiple ``@`` symbols, all but the most left
+            one will be treated as part of the regular ``back`` string.
+            This allows for usage of the symbol in descriptions, categories and tags.
+        """
+        result = string.split('@', 1)
+        length = len(result)
+        if length == 1:
+            front, back = result[0].strip(), None
+        else:
+            front, back = result
+            front, back = front.strip(), back.strip()
+        return (front, back)
+
+    def comma_split(string):
+        """
+        Split string at the most left comma.
+
+        Args:
+            string (str): String to be processed. At this stage this should
+                look something like ``<Category>, <Description>
+
+
+        Returns
+            tuple: (category_and_tags, description). Both substrings have their
+                leading/tailing whitespace removed.
+                ``category_and_tags`` may include >=0 tags indicated by a leading ``#``.
+                As we have used the most left ``,`` to separate both substrings that
+                means that categories and tags can not contain any ``,`` but the
+                description text may contain as many as wished.
+        """
+
+        result = tuple(string.split(',', 1))
+        length = len(result)
+        if length == 1:
+            category, description = result[0].strip(), None
+        else:
+            category, description = tuple(result)
+            category, description = category.strip(), description.strip()
+        return (category.strip(), description)
+
+    time_info, rest = time_helpers.extract_time_info(raw_fact)
+    activity_name, back = at_split(rest)
+
+    if back:
+        category_name, description = comma_split(back)
+    else:
+        category_name, description = None, None
+
+    return {
+        'timeinfo': time_info,
+        'category': category_name,
```

Comment:
This parsing method does not actually support extracting tag info. It did not before either.
I will add a inline comment pointing this out. Right now I do not want to spend time making it work with tags if we are likely to substitute the entire parsing function with a regex based approach anyway rather soon.